lib/bundler/source.rb

```
@@ -20,13 +20,21 @@ def version_message(spec)
         locked_spec = Bundler.locked_gems.specs.find {|s| s.name == spec.name }
         locked_spec_version = locked_spec.version if locked_spec
         if locked_spec_version && spec.version != locked_spec_version
-          message += Bundler.ui.add_color(" (was #{locked_spec_version})", :green)
+          message += Bundler.ui.add_color(" (was #{locked_spec_version})", earlier_version?(spec.version, locked_spec_version) ? :yellow : :green)
         end
       end
 
       message
     end
 
+    def earlier_version?(spec_version, locked_spec_version)
+      begin
+        Gem::Version.new(spec_version) < Gem::Version.new(locked_spec_version)
+      rescue
```

Preferred if you don't catch all exceptions